### PR TITLE
Task/admin form

### PR DESCRIPTION
**Pull request by AnatoliRub**
Implented admin form

### CODE REVIEW

File: youtube-client/src/app/admin/pages/admin-page/admin-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  getImageController() {
+    return new FormControl('', [Validators.required]);
+  }
+
+  getVideoLinkController() {
+    return new FormControl('', [Validators.required]);
+  }
+
+  get title() {
```
Comment (StanevichIvan) [inline youtube-client/src/app/admin/pages/admin-page/admin-page.component.ts:45]: please add return types for all functions

File: youtube-client/src/app/core/components/header/search-settings/search-settings.component.scss [hunk 2]
```diff
@@ -1,14 +1,17 @@
 @import 'src/styles.scss';
 
+:host {
+  margin-left: rem(14.5);
```
Comment (StanevichIvan) [inline youtube-client/src/app/core/components/header/search-settings/search-settings.component.scss:4]: is It required to have margin-left

File: youtube-client/src/app/shared/components/button/button.component.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<ng-container *ngTemplateOutlet="noImage() ? text : image"> </ng-container>
```
Comment (StanevichIvan) [inline youtube-client/src/app/shared/components/button/button.component.html:1]: might ngIfElse would work here?

**Review by StanevichIvan — APPROVED** [review]

File: youtube-client/src/app/shared/components/button/button.component.html (hunk 3 above)
Comment (AnatoliRub) [reply youtube-client/src/app/shared/components/button/button.component.html]: [f6811cc](https://github.com/AnatoliRub/rs-angular-itransition/pull/9/commits/f6811cc642e93c474d831b7a9a8087b5b2f07550)

File: youtube-client/src/app/core/components/header/search-settings/search-settings.component.scss (hunk 2 above)
Comment (AnatoliRub) [reply youtube-client/src/app/core/components/header/search-settings/search-settings.component.scss]: [eb580b7](https://github.com/AnatoliRub/rs-angular-itransition/pull/9/commits/eb580b7abb72891c903bdaff5e0823d4d04ab33b)

File: youtube-client/src/app/admin/pages/admin-page/admin-page.component.ts (hunk 1 above)
Comment (AnatoliRub) [reply youtube-client/src/app/admin/pages/admin-page/admin-page.component.ts]: [86b68f6](https://github.com/AnatoliRub/rs-angular-itransition/pull/9/commits/86b68f600b8801d247796b8a1d16f783e7914b6b)

File: youtube-client/src/app/youtube/pages/search-result-page/search-result-page.component.ts [hunk 4]
```diff
@@ -1,7 +1,10 @@
 import { Component } from '@angular/core';
+import { Router } from '@angular/router';
 import { Observable } from 'rxjs';
 import { SearchServiceService } from 'src/app/core/services/search-service.service';
+import { RoutesPath } from 'src/app/routes.enum';
 import { IFilterData } from 'src/types/filtering-criteria-types';
+import { faPlus } from '@fortawesome/free-solid-svg-icons';
```
Comment (StanevichIvan) [inline youtube-client/src/app/youtube/pages/search-result-page/search-result-page.component.ts:7]: what the reason of inporting icon into a component file. Might we could import font awesome module and use icon directly in template?

File: youtube-client/src/app/app-routing.module.ts [hunk 5]
```diff
@@ -5,20 +5,22 @@ import { RoutesPath } from './routes.enum';
...
   {
     path: RoutesPath.Youtube,
-    loadChildren: () => import('./youtube/youtube.module').then((m) => m.YoutubeModule),
+    loadChildren: (): Promise<any> =>
+      import('./youtube/youtube.module').then((m) => m.YoutubeModule),
   },
   {
     path: RoutesPath.Auth,
-    loadChildren: () => import('./auth/auth.module').then((m) => m.AuthModule),
+    loadChildren: (): Promise<any> => import('./auth/auth.module').then((m) => m.AuthModule),
```
[the remark this answers is not in the document]
Comment (AnatoliRub) [reply youtube-client/src/app/app-routing.module.ts]: [3977513](https://github.com/AnatoliRub/rs-angular-itransition/pull/9/commits/39775131b0664b025c591fa92d496a777252144e)

File: youtube-client/src/app/youtube/pages/search-result-page/search-result-page.component.ts (hunk 4 above)
Comment (AnatoliRub) [reply youtube-client/src/app/youtube/pages/search-result-page/search-result-page.component.ts:7]: It can't implement by this way.

**Pull request merged**